### PR TITLE
SWC-5784

### PR DESCRIPTION
**Pull request by nickgros**
https://user-images.githubusercontent.com/17580037/143269805-04b75edc-54cd-45e5-9e35-f97c8528a598.mov

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/jsinterop/EntityFinderProps.java [hunk 1]
```diff
@@ -52,7 +59,7 @@ public static EntityFinderProps create(
             List<EntityType> visibleTypesInList,
             List<EntityType> visibleTypesInTree,
             List<EntityType> selectableTypes,
-            String selectedCopy,
+            SelectedCopyHandler selectedCopy,
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/jsinterop/EntityFinderProps.java:62]: This is the copy in the entity finder that says "Selected" or "Destination". We can now pass a `fn(int selectedCount)` to say "4 Items Selected"

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableEntityWidget.java [hunk 2]
```diff
@@ -589,4 +604,23 @@ public void onAddToDownloadList() {
...
+	@Override
+	public DatasetEditorProps getItemsEditorProps() {
+		DatasetEditorProps props =
+				DatasetEditorProps.create(
+						entityBundle.getEntity().getId(),
+						() -> closeItemsEditor(),
+						() -> closeItemsEditor()
+				);
+		return props;
+	}
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableEntityWidget.java:617]: Presenter added so the view can get React component props from the model.

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableEntityWidget.java [hunk 3]
```diff
@@ -589,4 +604,23 @@ public void onAddToDownloadList() {
...
+
+	public void closeItemsEditor() {
+		isShowingItemsEditor = false;
+		actionMenu.setActionVisible(Action.EDIT_DATASET_ITEMS, true);
+		view.setItemsEditorVisible(false);
+		view.setQueryResultsVisible(true);
+		reconfigureState();
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableEntityWidget.java:624]: The only thing that *needs* to happen is calling `checkState` again, but I think it's safer to just capture the three methods called in the configure and call all of them.

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/schema/ColumnModelsWidget.java [hunk 4]
```diff
@@ -105,11 +105,11 @@ public void configure(EntityBundle bundle, boolean isEditable) {
...
 		viewer.configure(ViewType.VIEWER, this.isEditable);
-		// Certain rows in datasets may not be editable because they reference locked versions.
-		boolean isEditableView = isEditable && (bundle.getEntity() instanceof EntityView || bundle.getEntity() instanceof SubmissionView);
+		// We can get the default columns/annotations for views and datasets
+		boolean isEditableViewOrDataset = isEditable && (bundle.getEntity() instanceof EntityView || bundle.getEntity() instanceof SubmissionView || bundle.getEntity() instanceof Dataset);
 		tableType = TableType.getTableType(bundle.getEntity());
-		editor.setAddDefaultViewColumnsButtonVisible(isEditableView);
-		editor.setAddAnnotationColumnsButtonVisible(isEditableView);
+		editor.setAddDefaultViewColumnsButtonVisible(isEditableViewOrDataset);
+		editor.setAddAnnotationColumnsButtonVisible(isEditableViewOrDataset);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/schema/ColumnModelsWidget.java:112]: Addresses [SWC-5857](https://sagebionetworks.jira.com/browse/SWC-5857) -- buttons to add default columns or annotation columns were missing for datasets. I think in my first pass at this, I thought this was the query results editor, so the comment I left was irrelevant.

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**